app/src/main/java/com/mikhailovskii/weatherandroid/data/firebase/FirebaseModel.kt

```diff
@@ -0,0 +1,188 @@
+package com.mikhailovskii.weatherandroid.data.firebase
+
+import com.google.firebase.firestore.FirebaseFirestore
+import com.mikhailovskii.weatherandroid.data.entities.StickerPack
+import com.mikhailovskii.weatherandroid.data.entities.User
+import com.twitter.sdk.android.core.Result
+import timber.log.Timber
+
+class FirebaseModel {
+
+    private val db = FirebaseFirestore.getInstance()
+
+    fun getStickersCollectionList(callback: FirebaseDataCallback?) {
+        db.collection("stickers").get().addOnSuccessListener { result ->
+            val stickerPackList = ArrayList<StickerPack>()
+
+            for (document in result) {
+                val stickerPack = document.toObject(StickerPack::class.java)
+                stickerPackList.add(stickerPack)
+            }
+
+            callback?.onFirebaseDataLoaded(stickerPackList)
+        }.addOnFailureListener {
+            callback?.onFirebaseDataFailed()
+        }
+
+    }
+
+    fun saveUserLocation(user: User?) {
+        db.collection(USERS_COLLECTION).whereEqualTo(LOGIN_FIELD, user?.login).get()
+            .addOnSuccessListener { databaseResult ->
+
+                val map: Map<String, String> = hashMapOf(LOCATION to (user?.location ?: ""))
+
+                for (document in databaseResult) {
+                    db.collection(USERS_COLLECTION).document(document.id).update(map)
+                }
+
+            }
+    }
+
+    fun baseLogInUser(login: String, password: String, callback: FirebaseLoginCallback?) {
+        db.collection(USERS_COLLECTION).get().addOnSuccessListener { result ->
+            for (document in result) {
+                var user = document.toObject(User::class.java)
+
+                if (user.login == login && user.password == password) {
+                    user = User(login = login, password = password)
+
+                    if (user.location?.isNotBlank() == true) {
+                        callback?.onLoggedInWithFilledInfo()
+                    } else {
+                        callback?.onLoggedInWithEmptyLocation()
+                    }
+                }
+            }
+        }.addOnFailureListener {
+            callback?.onLoginFailed()
+        }
+    }
+
+    fun logInWithTwitter(
+        result: Result<com.twitter.sdk.android.core.models.User>?,
+        user: User,
+        callback: FirebaseLoginCallback?
+    ) {
+        db.collection(USERS_COLLECTION).get().addOnSuccessListener { databaseResult ->
+            var isUserPresents = false
+
+            loop@ for (document in databaseResult) {
+                val databaseUser = document.toObject(User::class.java)
+
+                if (databaseUser.login == result?.data?.name
+                    && databaseUser.twitterKey == result?.data?.idStr
+                    && databaseUser.icon == result?.data?.profileImageUrl
+                ) {
+                    isUserPresents = true
+                    break@loop
+                }
+            }
```

Comment:
@SergeiMikhailovskii if you want to break the loop and you need a point break, you'd better use the following construction
```kotlin
databaseResult.forEach {
    if (condition) {
        return@forEach
    }
}
```
in this case you don't need to define a label so it looks slightly better